jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -175,6 +180,7 @@ public void doWrite(Server server, Query query, Iterable<Result> results) throws
 			for (Map.Entry<String, Object> values : resultValues.entrySet()) {
 				Object value = values.getValue();
 				if (isValidNumber(value)) {
+					typeNames = typeNamesAsTags ? typeNames : Collections.<String>emptyList();
```

Comment:
Since `typeNamesAsTags` and `typeNames` are invariants, this could be initialized in the constructor, and `typeNames`could be made final.

Side note, `typeNames` is an `ImmutableList`, so an empty immutable list should be created with [`ImmutableList.of()`](https://google.github.io/guava/releases/21.0/api/docs/com/google/common/collect/ImmutableList.html#of--).